src/lightning_app/components/auto_scaler.py

```diff
@@ -445,8 +446,14 @@ def workers(self) -> List[LightningWork]:
 
     def create_work(self) -> LightningWork:
         """Replicates a LightningWork instance with args and kwargs provided via ``__init__``."""
-        # TODO: Remove `start_with_flow=False` for faster initialization on the cloud
-        return self._work_cls(*self._work_args, **self._work_kwargs, start_with_flow=False)
+        self._work_kwargs.update(
+            dict(
+                port=find_free_network_port(),
```

Comment:
There is a PR #15819 which tries to avoid using taken port for an app URL at import time, but it doesn't solve the issue where multiple works get assigned to the same port number.

Reply:
You don't need to provide a port. If None is passed, it would find one using this exact same method.

Reply:
Thank you for pointing it out :) I didn't know that `LightningWork` has that already, and now I understand that the cause of the issue was the fact that PythonServer defaults to `7777` instead of `None` there. Will update this PR now

Reply:
@tchaton Could you have another look?